packages/core/src/odata-common/request-builder/batch/batch-response-parser.ts

```diff
@@ -122,7 +122,13 @@ export function splitChangeSetResponse(changeSetResponse: string): string[] {
  * @returns A list of sub responses represented as strings.
  */
 export function splitResponse(response: string, boundary: string): string[] {
-  const parts = response.split(`--${boundary}`).map(part => part.trim());
```

Comment:
This was causing the bug, it was trimming every part of a batch response, which is exactly what we want, except when the body is empty. When the body of the response is empty the `parseResponseBody()` function, which is used by the `parseResponseData()` function, is trying to parse the last line of the response with `JSON.parse()`, which results in an error because, with an empty body, the last line is not valid JSON, with my fix the last line is now an empty line if the response has no body.